src/utils/updateNodePosition.tsx

```diff
@@ -0,0 +1,81 @@
+import type { Node } from "@xyflow/react";
```

Comment:
Nitpick: The file name is `updateNodePosition.tsx` but the function name is `updateNodePositions`. The file should probably be renamed. Also the file should probably just be `.ts`.

`updateNodePosition.tsx` -> `updateNodePositions.ts`